### ISSUE TITLE
Intermittent 401s when obtaining token

### ISSUE THREAD

**Issue by btrautmann** [peter-murray/workflow-application-token-action#36]
We're seeing intermittent 401 status code responses when tryin to obtain a token:

```
Run peter-murray/workflow-application-token-action@8e1ba3bf1619726336414f1014e37f17fbadf1db
  with:
    application_id: ***
    application_private_key: ***
    permissions: contents:read
Error: Error: Failed to connect as application; status code: 401
'Issued at' claim ('iat') must be an Integer representing the time that the assertion was issued
Error: Failed to initialize GitHub Application connection using provided id and private key
```

**Runner details:**
```
Image: macos-13-arm64
  Version: 20231017.1
  Included Software: https://github.com/actions/runner-images/blob/macos-13-arm64/2023[10](https://github.com/Betterment/mobile/actions/runs/6620528616/job/18002684063#step:1:11)17.1/images/macos/macos-13-arm64-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/macos-13-arm64%2F20231017.1
```

Any idea what might be going on?

**Comment by peter-murray** [comment]
The defaults will attempt to connect to the GitHub Actions repository, has the application been installed in to the repository that the actions workflow is executing from?

**Comment by btrautmann** [comment]
> The defaults will attempt to connect to the GitHub Actions repository, has the application been installed in to the repository that the actions workflow is executing from?

Yes, the action worked completely fine and without issue (with the app installed to the repository) before switching to M1 runners. Swapping to M1 runners caused this to happen. FWIW we have switched over to `getsentry`'s implementation [here](https://github.com/getsentry/action-github-app-token) and no longer have issues. I say that not as a dig or anything, but just to help others find something that may work more reliably in the meantime and to compare implementations if you believe it may help. This action was hugely helpful to us in our CICD implementation and we're very thankful for that!
